hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/parse/QueryParser.java

```diff
@@ -75,7 +75,7 @@ public QueryParseResult parse(String sql) {
                 }
             }
         } catch (Exception e) {
-            throw QueryException.error(SqlErrorCode.PARSING, e.getMessage(), e);
+            throw QueryException.error(SqlErrorCode.PARSING, getTrimmedMessage(e), e);
```

Comment:
This catch block intercepts all exceptions produced during both parsing and semantical validation. The original issue is concerned with errors produced by the Calcite parser only. Now, if we have our own exception that has a line separator somewhere on purpose, it will be trimmed. We should limit the trimming to the parsing only (`parser.parseStmt()`).